store/src/java/com/zimbra/cs/filter/jsieve/EditHeaderExtension.java

```diff
@@ -564,4 +566,35 @@ private boolean matchValue(String regex, String value) {
         Matcher matcher = pattern.matcher(value);
         return matcher.matches();
     }
+
+    static public boolean isTolerableFormedMessage(ZimbraMailAdapter zma, String actionName, MimeMessage mm) {
```

Comment:
this method is not used anymore, so can be removed. In the edit header classes like AddHeader, you should check if parseStatus is equal to PARSESTATUS.MIMEMALFORMED, then return without performing any operation as we already know the mime is malformed.